src/sasa.h

```diff
@@ -0,0 +1,242 @@
+
+    void sync([[maybe_unused]] Space& other_space, [[maybe_unused]] const Change& change) {}
+
+    const std::vector<double>& getAreas() const { return areas; }
+
+    SASA(const json& j, [[maybe_unused]] Space& spc)
+        : SASA(spc, j.value("radius", 1.4) * 1.0_angstrom, j.value("slices", 20)) {}
+    SASA([[maybe_unused]] Space& spc, double probe_radius, int n_slices_per_atom)
+        : probe_radius(probe_radius), n_slices_per_atom(n_slices_per_atom) {}
+};
+
+// not used yet
+class SASACellList : public SASA {
+
+  private:
+    using CellList_t = CellList::CellListSpatial<CellList::CellListType<int, CellList::Grid::Grid3DPeriodic>>;
+    std::unique_ptr<CellList_t> cell_list; // pointer to cell list
+    double cell_size;                      // dimension of a single cell
+    using CellCoord = CellList::GridOf<CellList_t>::CellCoord;
+    std::vector<CellCoord> cell_offsets; // holds offsets which define a 3x3x3 cube around central cell
+
+  public:
+    SASACellList(const json& j, Space& spc)
+        : SASACellList(spc, j.value("radius", 1.4) * 1.0_angstrom, j.value("slices", 20)) {}
+
+    SASACellList(Space& spc, double probe_radius, int n_slices_per_atom) : SASA(spc, probe_radius, n_slices_per_atom) {}
+
+    void init(Space& spc) {
+
+        cell_offsets.clear();
+        for (auto i = -1; i <= 1; ++i) {
+            for (auto j = -1; j <= 1; ++j) {
+                for (auto k = -1; k <= 1; ++k) {
+                    cell_offsets.push_back(CellCoord(i, j, k));
+                }
+            }
+        }
+
+        double max_radius = 0.0;
+        for (const auto& particle : spc.particles) {
+            if (particle.traits().sigma > max_radius) {
+                max_radius = particle.traits().sigma;
+            }
+        }
+        cell_size = max_radius + 2.0 * probe_radius;
+        cell_list = std::make_unique<CellList_t>(spc.geometry.getLength(), cell_size);
+        for (auto& particle : spc.activeParticles()) {
+            const int i = static_cast<int>(std::addressof(particle) - std::addressof(spc.particles.at(0)));
+            spc.geometry.boundary(particle.pos);
+            cell_list->insertMember(i, particle.pos);
+        }
+
+        areas.resize(spc.particles.size());
+    }
+
+    SASA::NeighboursData calcNeighbourDataOfParticle(Space& spc, const int target_ind) {
+
+        SASA::NeighboursData neighbour;
+
+        const int i = target_ind;
+        const auto& particle_i = spc.particles.at(i);
+        neighbour.ind = i;
+        const auto rc_i = particle_i.traits().sigma * 0.5 + probe_radius;
+        const auto& center_cell = cell_list->getGrid().coordinatesAt(particle_i.pos);
+
+        for (const auto& offset : cell_offsets) {
+            const auto& neighbour_particle_inds = cell_list->getNeighborMembers(center_cell, offset);
+            for (const int j : neighbour_particle_inds) {
+
+                const auto& particle_j = spc.particles.at(j);
```

Comment:
I removed SASACellList class from sasa branch completely because it did not belong there anyway and added the proposed changes concerning the SASACellList class into sasa-cellist branch